hs628_tools/Preprocess1_module.py

```diff
@@ -9,8 +9,33 @@
 import matplotlib
 import matplotlib.pyplot as plt
 
+def main():
+    printIntro()
+    df1, col, col_cat = getInputs()
+    listgrp=coltuple(df1,col,col_cat)# helper function
+    colgroupbyclass=group_by(listgrp,1) # helper function
+    print(summarystat(colgroupbyclass))
 
-df1 = pd.read_csv("C:/Users/Nitie/Desktop/health_informatics/MSHI_CLASSES/summer_2018/650/project data/parkinsons_train_set.csv")
+def printIntro():
+               Print("The purpose of this program is to visualize some descriptive statistics grouped by column")
+
+def getInputs() :
+        """
+           Build a tuple from the column and class label
+            -----------
+            Parameters:
+            Prompt User for name categorical column
+            User input for name of target column
+            Prompt user for data frame
+            -----------
+            Returns:
+            Variables representing categorical column,target column and dataframe
+        """
+               #Returns the three input parameters
+               col_cat=(input("what is col_cat name ?"))
+               col=float(input("what is the col name ?"))
```

Comment:
Hi Nitie, I just tried running your work for review on my PyCharm and got the error message as below:
File "<input>", line 35
    col_cat=(input("what is col_cat name ?"))
    ^
IndentationError: unexpected indent

Just wanted to suggest you revisit your work to confirm debugging

Reply:
![indentationerror](https://user-images.githubusercontent.com/35585357/43371845-c3a34b64-93d4-11e8-8057-d6d5d44f4ba1.png)
Hope this attached hleps!